x-pack/plugins/alerts/server/alerts_client/alerts_client.ts

```diff
@@ -396,6 +400,77 @@ export class AlertsClient {
     };
   }
 
+  public async getHealth(): Promise<AlertsHealth> {
+    const { saved_objects: data } = await this.unsecuredSavedObjectsClient.find<RawAlert>({
+      filter: 'alert.attributes.executionStatus.status:error',
+      fields: ['executionStatus'],
+      type: 'alert',
+    });
+
+    const healthStatuses = data.reduce(
+      (prevItem: AlertsHealth, item: SavedObjectsFindResult<RawAlert>) => {
+        switch (item.attributes.executionStatus.error?.reason) {
+          case AlertExecutionStatusErrorReasons.Decrypt:
+            prevItem.decryptionHealth = {
+              status: HealthStatus.Warning,
+              timestamp: item.attributes.executionStatus.lastExecutionDate,
+            };
+            break;
+          case AlertExecutionStatusErrorReasons.Execute:
+            prevItem.executionHealth = {
+              status: HealthStatus.Warning,
+              timestamp: item.attributes.executionStatus.lastExecutionDate,
+            };
+            break;
+          case AlertExecutionStatusErrorReasons.Read:
+            prevItem.readHealth = {
+              status: HealthStatus.Warning,
+              timestamp: item.attributes.executionStatus.lastExecutionDate,
+            };
+            break;
+        }
+        return prevItem;
+      },
+      {
+        decryptionHealth: {
+          status: HealthStatus.OK,
+          timestamp: '',
+        },
+        executionHealth: {
+          status: HealthStatus.OK,
+          timestamp: '',
+        },
+        readHealth: {
+          status: HealthStatus.OK,
+          timestamp: '',
+        },
+      }
+    );
+
+    const { saved_objects: noErrorData } = await this.unsecuredSavedObjectsClient.find<RawAlert>({
+      filter:
+        'alert.attributes.executionStatus.status:ok or alert.attributes.executionStatus.status:active or alert.attributes.executionStatus.status:pending',
```

Comment:
I wonder if this would be better (and possible) by changing to 

```
not alert.attributes.executionStatus.status:error
```

In case we add more statuses later.